nilearn/interfaces/fmriprep/load_confounds_components.py

```diff
@@ -26,6 +28,114 @@
 )
 
 
+def _tedana_strategy(classification, mixing, metrics):
+    """
+    Auxiliary functionality to extract confounds based on the strategy.
+
+    Parameters
+    ----------
+    classification : :obj:`list`
+        list of classifications to be returned from the mixing file
+
+    mixing : :obj:`pd.DataFrame`
+        pandas dataframe with the mixing file.
+
+    metrics : :obj:`pd.DataFrame`
+        pandas dataframe with the metrics file.
+
+    Returns
+    -------
+    pandas.DataFrame
+        DataFrame of TEDANA regressors.
+    """
+    # 1. Get selected components from status table
+    selected_metric_components = metrics[
+        metrics["classification"].isin(classification)
+    ]
+
+    #############################################################
+    # tedana versions like 24.0.2 do not have matching component
+    # names from the mixing and status tables. For backwards
+    # compatibility we will normalize the component names to
+    # match the mixing file rejected component
+    # names: ICA_04 -> ICA_4
+    #############################################################
+
+    # 3. Normalize mixing file column names and build a mapping
+    original_columns = mixing.columns.tolist()
+
+    # Map: normalized_name → original_name
+    column_mapping = {
+        re.sub(r"ICA_0*(\d+)$", lambda m: f"ICA_{int(m.group(1))}", col): col
+        for col in original_columns
+    }
+
+    # Apply normalized names to columns
+    mixing.columns = list(column_mapping.keys())
+
+    # Build classification lookup from normalized component names
+    classification_lookup = {
+        re.sub(
+            r"ICA_0*(\d+)$",
+            lambda m: f"ICA_{int(m.group(1))}",
+            row["Component"],
+        ): row["classification"]
+        for _, row in selected_metric_components.iterrows()
+    }
+
+    # 4. Select matched columns (normalized)
+    matched_components = [
+        c for c in classification_lookup if c in mixing.columns
+    ]
+    selected = mixing[sorted(matched_components)]
+
+    # 5. Rename columns using classification + original name
+    renamed_columns = {
+        col: f"{classification_lookup[col]}_{column_mapping[col]}"
+        for col in selected.columns
+    }
+    load_confounds = selected.rename(columns=renamed_columns)
+
+    return load_confounds
```

Comment:
naming a variable with the same name of an existing function is confusing. 
```suggestion
    return selected.rename(columns=renamed_columns)
```